### PR TITLE
allow using lightning/PL & v2.0.0rc

### PR DESCRIPTION
**Pull request by Borda**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- Did you make sure to update the docs?
- [x] Did you all existing and newly added tests pass locally?

</details>

## What does this PR do?

it shall be fully compatible with both packages lightning and pytorch-lightning.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

### CODE REVIEW

File: tests/test_strategy_cpu.py [hunk 1]
```diff
@@ -135,8 +146,9 @@ def training_step(self, batch, batch_idx):
                     res["test_tensor"].item() == hvd.size()
                 ), "Result-Log does not work properly with Horovod and Tensors"
 
-            def training_epoch_end(self, outputs) -> None:
-                assert len(outputs) == 0
+            # ToDo: find alternative for this check
+            # def training_epoch_end(self, outputs) -> None:
+            #     assert len(outputs) == 0
```
Comment (awaelchli) [inline tests/test_strategy_cpu.py:152]: There is no alternative. You can just safely remove it.

File: tests/test_strategy_cpu.py [hunk 2]
```diff
@@ -79,6 +87,9 @@ def test_implicit(tmpdir):
     _run_horovod(trainer_options)
 
 
+@pytest.mark.xfail(
+    raises=RuntimeError, reason="Training with multiple optimizers is only supported with manual optimization"
+)
```
Comment (awaelchli) [inline tests/test_strategy_cpu.py:93]: The BasicGAN needs to be rewritten with manual optimization. You can copy paste BasicGAN from lightning under tests_pytorch/helpers/advanced_models.py

Comment (Borda) [reply tests/test_strategy_cpu.py:93]: yes, that is why I will make it as xfail not just silent drop :)

Comment (awaelchli) [reply tests/test_strategy_cpu.py:93]: I wouldn't make it as xfail. The simplest is to copy-past the BasicGAN over to tests/helpers.

Comment (Borda) [reply tests/test_strategy_cpu.py:93]: thete is already one `xfail` but pytest does not allow mark one tests with two so far I know...

Comment (Borda) [reply tests/test_strategy_cpu.py:93]: found it... https://stackoverflow.com/a/38403866/4521646
but still we do not expect user using this with aligning below 2.0, right?

File: src/lightning_horovod/strategy.py [hunk 3]
```diff
@@ -18,21 +18,39 @@
...
-from pytorch_lightning.plugins.precision import PrecisionPlugin
-from pytorch_lightning.strategies.parallel import ParallelStrategy
-from pytorch_lightning.strategies.strategy import TBroadcast
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from pytorch_lightning.utilities.rank_zero import rank_zero_only
+from lightning_utilities import module_available
 from torch import Tensor
 from torch.optim import Optimizer
 
+if module_available("lightning"):
```
Comment (justusschock) [inline src/lightning_horovod/strategy.py:25]: this still has the issue that you cannot use the strategy with `pytorch_lightning` if `lightning` is installed, right?

**Review by justusschock — APPROVED** [review]

File: src/lightning_horovod/strategy.py (hunk 3 above)
Comment (Borda) [reply src/lightning_horovod/strategy.py:25]: Could be but this issue would be for any user who has installed both, right?
anyway I would land this and wait is someone cross the mentioned issue :otter:

**Pull request merged**